sycl/include/CL/sycl/usm.hpp

```diff
@@ -158,7 +158,8 @@ T *malloc_device(
     const property_list &PropList = {},
     const detail::code_location CodeLoc = detail::code_location::current()) {
   return static_cast<T *>(
-      malloc_device(Count * sizeof(T), Dev, Ctxt, PropList, CodeLoc));
+      aligned_alloc_device(std::max(alignof(T), alignof(std::max_align_t)),
```

Comment:
@gmlueck , after reading your comment in the llvm-test-suite PR I think this is overly pessimistic and should be just `alignof(T)` without any `std::max`. Would you agree?

Reply:
Yes, you are correct.  In fact, the [spec ](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_usm_allocations)says this:

> No alignment parameter; Templated on allocation type `T`
> Pointer is suitably aligned for an object of type `T`.

Thanks for noticing.